Build failure: nfd

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

```
❯ nix shell nixpkgs#nfd
error: builder for '/nix/store/hv7nla3yinz5dh1246vn20gplznr3134-nfd-24.07.drv' failed with exit code 1;
       last 25 log lines:
       > ../websocketpp/websocketpp/transport/asio/endpoint.hpp:1122: error: ‘m_io_service’ was not declared in this scope
       >  1122 |         ec = tcon->init_asio(m_io_service);
       > 
       > In file included from include/boost/multi_index_container.hpp:59,
       >                  from include/boost/property_tree/ptree.hpp:21,
       >                  from ../daemon/common/config-file.hpp:31,
       >                  from ../daemon/nfd.hpp:29,
       >                  from ../daemon/main.cpp:26:
       > include/boost/multi_index/detail/serialization_version.hpp:37:74: warning: extra ‘;’ after in-class function definition [-Wextra-semi]
       > In file included from include/boost/asio/basic_signal_set.hpp:25,
       >                  from include/boost/asio/signal_set.hpp:19,
       >                  from ../daemon/main.cpp:36:
       > include/boost/asio/detail/signal_set_service.hpp:249:11: warning: redundant class-key ‘class’ in reference to ‘class boost::asio::detail::scheduler’ [-Wredundant-tags]
       > In file included from include/boost/asio/detail/signal_set_service.hpp:291:
       > include/boost/asio/detail/impl/signal_set_service.ipp:54:3: warning: redundant class-key ‘class’ in reference to ‘class boost::asio::detail::signal_set_service’ [-Wredundant-tags]
       > In file included from ../daemon/main.cpp:40:
       > include/boost/program_options/parsers.hpp:223:26: warning: redundant enum-key ‘enum’ in reference to ‘enum boost::program_options::collect_unrecognized_mode’ [-Wredundant-tags]
       > In file included from include/boost/program_options/parsers.hpp:299:
       > include/boost/program_options/detail/parsers.hpp:116:26: warning: redundant enum-key ‘enum’ in reference to ‘enum boost::program_options::collect_unrecognized_mode’ [-Wredundant-tags]
       > 
       > Waf: Leaving directory `/build/source/build'
       > Build failed
       >  -> task in 'daemon-objects' failed with exit status 1 (run with -v to display more information)
       >  -> task in 'daemon-objects' failed with exit status 1 (run with -v to display more information)
       >  -> task in 'daemon-objects' failed with exit status 1 (run with -v to display more information)
       For full logs, run 'nix log /nix/store/hv7nla3yinz5dh1246vn20gplznr3134-nfd-24.07.drv'.
```

Full build log: 
[build-log.txt](https://github.com/user-attachments/files/18619615/build-log.txt)

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/285949997

### Relevant log output

```console
In file included from ../websocketpp/websocketpp/transport/asio/endpoint.hpp:32,
                 from ../websocketpp/websocketpp/config/asio_no_tls_client.hpp:32,
                 from ../daemon/face/websocketpp.hpp:39,
                 from ../daemon/face/websocket-channel.hpp:30,
                 from ../daemon/face/websocket-channel.cpp:26:
../websocketpp/websocketpp/transport/asio/connection.hpp:89: error: ‘io_service’ in namespace ‘websocketpp::lib::asio’ does not name a type
   89 |     typedef lib::asio::io_service * io_service_ptr;
../websocketpp/websocketpp/transport/asio/connection.hpp:91: error: ‘io_service’ is not a member of ‘websocketpp::lib::asio’
   91 |     typedef lib::shared_ptr<lib::asio::io_service::strand> strand_ptr;
../websocketpp/websocketpp/transport/asio/connection.hpp:91: error: template argument 1 is invalid
../websocketpp/websocketpp/transport/asio/connection.hpp:461: error: ‘io_service_ptr’ has not been declared
  461 |     lib::error_code init_asio (io_service_ptr io_service) {
../websocketpp/websocketpp/transport/asio/connection.hpp:1175: error: ‘io_service_ptr’ does not name a type
 1175 |     io_service_ptr  m_io_service;
../websocketpp/websocketpp/transport/asio/connection.hpp: In member function ‘websocketpp::transport::asio::connection<config>::timer_ptr websocketpp::transport::asio::connection<config>::set_timer(long int, websocketpp::transport::timer_handler)’:
../websocketpp/websocketpp/transport/asio/connection.hpp:316: error: ‘m_io_service’ was not declared in this scope; did you mean ‘m_is_server’?
  316 |                 *m_io_service,
../websocketpp/websocketpp/transport/asio/connection.hpp:321: error: base operand of ‘->’ is not a pointer
  321 |             new_timer->async_wait(m_strand->wrap(lib::bind(
../websocketpp/websocketpp/transport/asio/connection.hpp: In member function ‘std::error_code websocketpp::transport::asio::connection<config>::init_asio(int)’:
../websocketpp/websocketpp/transport/asio/connection.hpp:462: error: ‘m_io_service’ was not declared in this scope; did you mean ‘io_service’?
  462 |         m_io_service = io_service;
../websocketpp/websocketpp/transport/asio/connection.hpp:465: error: request for member ‘reset’ in ‘((websocketpp::transport::asio::connection<config>*)this)->websocketpp::transport::asio::connection<config>::m_strand’, which is of non-class type ‘websocketpp::transport::asio::connection<config>::strand_ptr’ {aka ‘int’}
  465 |             m_strand.reset(new lib::asio::io_service::strand(*io_service));
../websocketpp/websocketpp/transport/asio/connection.hpp:465: error: ‘io_service’ in namespace ‘websocketpp::lib::asio’ does not name a type
../websocketpp/websocketpp/transport/asio/connection.hpp: In member function ‘void websocketpp::transport::asio::connection<config>::handle_post_init(timer_ptr, websocketpp::transport::init_handler, const std::error_code&)’:
../websocketpp/websocketpp/transport/asio/connection.hpp:576: error: ‘using std::__shared_ptr_access<boost::asio::basic_waitable_timer<std::chrono::_V2::steady_clock>, __gnu_cxx::_S_atomic, false, false>::element_type = class boost::asio::basic_waitable_timer<std::chrono::_V2::steady_clock>’ {aka ‘class boost::asio::basic_waitable_timer<std::chrono::_V2::steady_clock>’} has no member named ‘expires_from_now’
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.11, NixOS, 25.05 (Warbler), 25.05.20250126.852ff1d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/etc/nix/path/nixpkgs`

### Notify maintainers

@bertof 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/nf/nfd/package.nix
+++ b/pkgs/by-name/nf/nfd/package.nix
@@ -1,7 +1,7 @@
 {
   lib,
   stdenv,
-  boost,
+  boost186,
   fetchFromGitHub,
   libpcap,
   ndn-cxx,
@@ -44,8 +44,8 @@ stdenv.mkDerivation rec {
     ++ lib.optional withWebSocket websocketpp
     ++ lib.optional withSystemd systemd;
   wafConfigureFlags = [
-    "--boost-includes=${boost.dev}/include"
-    "--boost-libs=${boost.out}/lib"
+    "--boost-includes=${boost186.dev}/include"
+    "--boost-libs=${boost186.out}/lib"
   ] ++ lib.optional (!withWebSocket) "--without-websocket";
 
   meta = with lib; {

